### ISSUE TITLE
[Clang] Does not catch undefined behavior in post-increment during constant evaluation

### ISSUE THREAD

**Issue by shafik** [llvm/llvm-project#82043]
As noted by [cwg2855](https://github.com/cplusplus/CWG/issues/479) the following code invoked undefined behavior:

```cpp
#include <cstdint>

int8_t constexpr f() {
  int8_t x = 127;
  return x++;
}

int8_t constexpr x = f();
```

We should catching this during constant evaluation, so the code above should be ill-formed, but we currently don't: https://godbolt.org/z/PY465bTzo

**Comment by shafik** [comment]
I misunderstood cwg2855, the goal is make this not undefined behavior. So this will likely be closed as not needed once the issues is closed.

**Comment by efriedma-quic** [comment]
The standard was revised; I think this is not undefined behavior according to the current text.
